templates/template_components.adoc

```diff
@@ -0,0 +1,343 @@
+<4> xref:../../dev_guide/templates/template_components.adoc#dev-guide-templates-components-object-list[Objects] created when the template is instantiated.
+
+
+
+[[dev-guide-templates-components-description]]
+== Description
+
+The template description reports what the template does and helps you
+find a the right template when searching in the web console. Additional metadata beyond the
+template name is optional, but useful to have. In addition to general
+descriptive information, the metadata also includes a set of tags. Useful tags
+include the name of the language the template is related to (for example,
+*java*, *php*, *ruby*, and so forth).
+
+.Template Description Metadata
+[source,yaml]
+----
+kind: Template
+apiVersion: v1
+metadata:
+  name: cakephp-mysql-example <1>
+  annotations:
+    openshift.io/display-name: "CakePHP MySQL Example (Ephemeral)" <2>
+    description: >-
+      An example CakePHP application with a MySQL database. For more information
+      about using this template, including {product-title} considerations, see
+      https://github.com/openshift/cakephp-ex/blob/master/README.md.
+
+
+      WARNING: Any data stored will be lost upon pod destruction. Only use this
+      template for testing." <3>
+    template.openshift.io/long-description: >-
+      This template defines resources needed to develop a CakePHP application,
+      including a build configuration, application deployment configuration, and
+      database deployment configuration.  The database is stored in
+      non-persistent storage, so this configuration should be used for
+      experimental purposes only. <4>
+    tags: "quickstart,php,cakephp" <5>
+    iconClass: icon-php <6>
+    template.openshift.io/provider-display-name: "Red Hat, Inc." <7>
+    template.openshift.io/documentation-url: "https://github.com/openshift/cakephp-ex" <8>
+    template.openshift.io/support-url: "https://access.redhat.com" <9>
+message: "Your admin credentials are ${ADMIN_USERNAME}:${ADMIN_PASSWORD}" <10>
+----
+<1> The unique internal name for the template.
+<2> A brief, user-friendly name, which can be employed by user interface.
+<3> A description of the template. Include enough detail that the user will
+understand what is being deployed and any caveats they need to know before
+deploying. It should also provide links to additional information, such as a
+*_README_* file. Newlines can be included to create paragraphs.
+<4> Additional template description. For example, this can be displayed by the service
+catalog.
+<5> Tags to be associated with the template for searching and grouping. Add tags
+that will include it into one of the provided catalog categories. Refer to the
+`id` and `categoryAliases` in `CATALOG_CATEGORIES` in the console's
+link:https://github.com/openshift/origin-web-console/blob/master/app/scripts/constants.js[constants
```

Comment:
This is no longer the right file. It should be `categories` in

https://github.com/openshift/origin-web-catalog/blob/master/src/constants.ts